redis/connection.py

```diff
@@ -407,7 +412,10 @@ def on_connect(self):
 
     def disconnect(self, *args):
         "Disconnects from the Redis server"
-        self._parser.on_disconnect()
+        try:
+            self._parser.on_disconnect()
+        except BaseException as be:
+            logger.exception('Exception calling parser on_disconnect()', exc_info=be)
```

Comment:
Should we still raise here after logging?

Reply:
Yeah, good call